### PR TITLE
Source Github Singer

### PR DESCRIPTION
**Pull request by cgardens**
## What
* Implemented github source pattern matching off of work done for the stripe source. 
* Going to figure out the integration test story now using this as a guinea pig.

## Checklist
- [ ] *Unit Tests*
- [ ] *Integration Tests*

## Recommended reading order
1. `airbyte-integrations/connectors/source-github-singer/source_github_singer/source.py`
1. the rest

### CODE REVIEW

File: airbyte-integrations/connectors/source-github-singer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM airbyte/integration-base-singer:dev
+
+RUN apt-get update && apt-get install -y \
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-github-singer/Dockerfile:3]: @jrhizor are these the same for all singer / postgres integrations. could we move this into the base? happy to just do it in this PR if it's the right thing to do.

Comment (jrhizor) [reply airbyte-integrations/connectors/source-github-singer/Dockerfile:3]: I don't think these should be necessary for most integrations. They should be only included if necessary.

File: tools/bin/ci_credentials.sh [hunk 2]
```diff
@@ -7,3 +7,7 @@ echo "$BIGQUERY_INTEGRATION_TEST_CREDS" > airbyte-integrations/connectors/destin
 
 mkdir airbyte-integrations/connectors/source-stripe-singer/secrets/
 echo "$STRIPE_INTEGRATION_TEST_CREDS" > airbyte-integrations/connectors/source-stripe-singer/secrets/config.json
+
+mkdir airbyte-integrations/connectors/source-github-singer/secrets/
+# pull sample config. add in the access key. write to secrets.
+jq .access_token=$GH_INTEGRATION_TEST_CREDS airbyte-integrations/connectors/source-github-singer/config.sample.json > airbyte-integrations/connectors/source-github-singer/secrets/config.json
```
Comment (jrhizor) [inline tools/bin/ci_credentials.sh:13]: I like this.

**Review by jrhizor — APPROVED** [review]

**Review by michel-tricot — APPROVED** [review]

File: airbyte-integrations/bases/base-singer/base_singer/singer_helpers.py [hunk 3]
```diff
@@ -134,7 +134,8 @@ def create_singer_catalog_with_selection(masked_airbyte_catalog, discovered_sing
         for singer_stream in discovered_singer_catalog.get("streams"):
             if singer_stream.get("stream") in stream_to_airbyte_schema:
                 new_metadatas = []
-
+                # support old style catalog.
+                singer_stream["schema"]["selected"] = True
```
Comment (cgardens) [inline airbyte-integrations/bases/base-singer/base_singer/singer_helpers.py:138]: @jrhizor - i think this integration uses old style singer. I think this is the right place to fix this. i'm surprised i thought we had other integrations that did this.

**Pull request merged**